examples/step-6/doc/results.dox

```diff
@@ -484,6 +484,42 @@ step-49, step-53, and step-54, among other tutorial programs that cover
 the issue. Information on curved domains can also be found in the
 documentation module on @ref manifold "Manifold descriptions".
 
+Why does it make sense to choose a mesh that tracks the internal
+interface? There are a number of reasons, but the most essential one
+comes down to what we actually integrate in our bilinear
+form. Conceptually, we want to integrate the term $A_{ij}^K=\int_K
+a(\mathbf x) \varphi_i(\mathbf x) \varphi_j(\mathbf x) ; dx$ as the
+contribution of cell $K$ to the matrix entry $A_{ij}$. We can not
+compute it exactly and have to resort to quadrature. We know that
+quadrature is accurate if the integrand is smooth. That is because
+quadrature in essence computes a polynomial approximation to the
+integrand that coincides with the integrand in the quadrature points,
+and then computes the volume under this polynomial as an approximation
+to the volume under the original integrand. This polynomial
+interpolant is accurate if the integrand is smooth on a cell, but it
+is usually rather inaccurate if the integrand is discontinuous on a
+cell.
+
+Consequently, it is worthwhile to align cells in such a way that the
+interfaces where the coefficient is discontinuous are aligned with
+cell interfaces. This way, the coefficient is constant on each cell,
+the integrand is smooth on each cell, its polynomial approximation is
+accurate, and the quadrature approximation of the integral is accurate
+as well. Note, that such an alignment is rather common in many
+practical cases, so deal.ii provides a number of functions
+(e.g. CellAccessor::material_id() and so on) to manage such a case.
+Refer to  step-28 as an advanced usage example.
+
+Finally, let us consider the case of a coefficient with smooth and
+non-uniform distribution in space. We can repeat once again all the
+discussion on the representation of such a function with the
+quadrature. So, to simulate it accurately you have basically few
+options: you can reduce the cell size and/or increase the order of the
+polynomial used in the quadrature formula and/or select a more
+appropriate quadrature formula. The key is to provide the best fit of
+the coefficient dependence with the quadrature polynomial interpolant,
+this will provide an accurate mapping of your model to the FEM
```

Comment:
I`ve merged most of the previous comments, however, the last three are simply not really correct.  It is about space dependence (added "space" to the text) of the coefficient and how it is taken into account in quadrature integration (via polynomial approximation, so "and" is wrong). Finite element space refers to the solution, not to the coefficient, so this is completely wrong (however, weak form integration is a part of any FEM formulation).